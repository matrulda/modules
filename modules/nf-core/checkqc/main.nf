process CHECKQC {
    label 'process_single'

    conda "bioconda::checkqc=3.6.1"
    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'https://depot.galaxyproject.org/singularity/checkqc:3.6.1--py_0':
        'biocontainers/checkqc:3.6.1--py_0' }"

    input:
    path(run_dir)
    path(checkqc_config)

    output:
    path "*checkqc_report.json", emit: report
    path "versions.yml"        , emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def args = task.ext.args ?: ''
    def config = checkqc_config ? "--config $checkqc_config" : ''

    """
    checkqc \
        $args \
        $config \
        --json \
        $run_dir >> checkqc_report.json || test -f "checkqc_report.json"

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        checkqc: \$( checkqc --version | sed -e "s/checkqc, version //g" )
    END_VERSIONS
    """

    stub:
    def args = task.ext.args ?: ''
    """
    touch checkqc_report.json

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        checkqc: \$( checkqc --version | sed -e "s/checkqc, version //g" )
    END_VERSIONS
    """
}
