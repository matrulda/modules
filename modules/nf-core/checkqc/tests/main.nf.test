nextflow_process {

    name "Test Process CHECKQC"
    script "modules/nf-core/checkqc/main.nf"
    process "CHECKQC"
    tag "checkqc"
    tag "modules"
    tag "modules_nfcore"

    test("Run checkqc without error") {
        config "./test_checkqc.config"

        setup {
            run("UNTAR") {
                script "modules/nf-core/untar/main.nf"
                process {
                    """
                    input[0] = Channel.of([ [], file(params.test_data['homo_sapiens']['illumina']['test_interop_bcl2fastqstats_flowcell'], checkIfExists: true)])
                    """
                }
            }
        }

        when {
            process {
                """
                    input[0] = UNTAR.out.untar.map{ it[1] }
                    input[1] = []
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }

    }

}
