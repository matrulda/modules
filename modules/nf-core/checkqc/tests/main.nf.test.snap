{
    "homo_sapiens illumina test_interop_bcl2fastqstats_flowcell - QC FAIL": {
        "content": [
            {
                "0": [
                    "checkqc_report.json:md5,0f27336ae332afe0791d0d3be0edb976"
                ],
                "1": [
                    "versions.yml:md5,c37381fc450f4ed00da2987430db8177"
                ],
                "report": [
                    "checkqc_report.json:md5,0f27336ae332afe0791d0d3be0edb976"
                ],
                "versions": [
                    "versions.yml:md5,c37381fc450f4ed00da2987430db8177"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.04.3"
        },
        "timestamp": "2024-07-17T17:03:04.717898665"
    },
    "homo_sapiens illumina test_interop_bcl2fastqstats_flowcell - QC OK": {
        "content": [
            {
                "0": [
                    "checkqc_report.json:md5,c69cb912a8c5bf53544764bd5bd73ae5"
                ],
                "1": [
                    "versions.yml:md5,c37381fc450f4ed00da2987430db8177"
                ],
                "report": [
                    "checkqc_report.json:md5,c69cb912a8c5bf53544764bd5bd73ae5"
                ],
                "versions": [
                    "versions.yml:md5,c37381fc450f4ed00da2987430db8177"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.04.3"
        },
        "timestamp": "2024-07-17T17:02:46.252509937"
    }
}